src/ripple/ledger/ApplyView.h

```diff
@@ -212,6 +213,67 @@ class ApplyView
         std::uint32_t cur, std::uint32_t next)
     {};
 
+    /** Add an entry to a directory
+
+     @param directory the base of the directory
+     @param key the entry to insert
+     @param strictOrder keep entries in order of insertion
```

Comment:
This function is actually giving me heartburn. I dislike functions that take a `bool` and implement two different kinds of behavior based on that. Very error prone.

I seriously considered having two functions. Something like this:

```
boost::optional<std::uint64_t>
ApplyView::dirInsertSorted (
    Keylet const& directory,
    uint256 const& key,
    std::function<void (std::shared_ptr<SLE> const&)> describe)
{
    // dirInsert is a private function:
    return dirInsert (directory, key, describe, 
        [](STVector256& indexes, uint256 const& key)
        {
            // We can't be sure if this page is already sorted because
            // it may be a legacy page we haven't yet touched. Take
            // the time to sort it.
            std::sort (indexes.begin(), indexes.end());

            auto pos = std::lower_bound(indexes.begin(), indexes.end(), key);

            if (pos != indexes.end() && key == *pos)
                LogicError ("dirInsert: double insertion");

            indexes.insert (pos, key);
        });
}

boost::optional<std::uint64_t>
ApplyView::dirInsertUnsorted (
    Keylet const& directory,
    uint256 const& key,
    std::function<void (std::shared_ptr<SLE> const&)> describe)
{
    // dirInsert is a private function:
    return dirInsert (directory, key, describe, 
        [](STVector256& indexes, uint256 const& key)
        {
            if (std::find(indexes.begin(), indexes.end(), key) != indexes.end())
                LogicError ("dirInsert: double insertion");

            indexes.push_back(key);
        }
}
```